harness/determined/common/api/_session.py

```diff
@@ -22,69 +22,22 @@
 )
 
 
-def _do_request(
-    method: str,
-    host: str,
-    path: str,
-    max_retries: Optional[GeneralizedRetry],
-    params: Optional[Dict[str, Any]] = None,
-    json: Any = None,
-    data: Optional[str] = None,
-    headers: Optional[Dict[str, str]] = None,
-    cert: Optional[certs.Cert] = None,
-    timeout: Optional[Union[Tuple, float]] = None,
-    stream: bool = False,
-) -> requests.Response:
-    # Allow the json to come pre-encoded, if we need custom encoding.
-    if json is not None and data is not None:
-        raise ValueError("json and data must not be provided together")
-
-    if json:
-        data = det.util.json_encode(json)
-
-    relpath = path.lstrip("/")
-
-    try:
-        r = det_requests.request(
-            method,
-            f"{host}/{relpath}",
-            params=params,
-            data=data,
-            headers=headers,
-            verify=cert.bundle if cert else None,
-            stream=stream,
-            timeout=timeout,
-            server_hostname=cert.name if cert else None,
-            max_retries=max_retries,
-        )
-    except requests.exceptions.SSLError:
-        raise
-    except requests.exceptions.ConnectionError as e:
-        raise errors.MasterNotFoundException(str(e))
-    except requests.exceptions.RequestException as e:
-        raise errors.BadRequestException(str(e))
-
-    def _get_error_str(r: requests.models.Response) -> str:
-        try:
-            json_resp = _json.loads(r.text)
-            mes = json_resp.get("message")
-            if mes is not None:
-                return str(mes)
-            # Try getting GRPC error description if message does not exist.
-            return str(json_resp.get("error").get("error"))
-        except Exception:
-            return ""
-
-    if r.status_code == 403:
-        raise errors.ForbiddenException(message=_get_error_str(r))
-    if r.status_code == 401:
-        raise errors.UnauthenticatedException()
-    elif r.status_code == 404:
-        raise errors.NotFoundException(_get_error_str(r))
-    elif r.status_code >= 300:
-        raise errors.APIException(r)
-
-    return r
+def _make_requests_session(
+    server_hostname: Optional[str] = None,
+    verify: Optional[Union[str, bool]] = None,
+    max_retries: Optional[GeneralizedRetry] = None,
+    headers: Optional[Dict[str, Any]] = None,
+) -> requests.Session:
+    requests_session = requests.Session()
+    requests_session.mount(
+        "https://", _HTTPSAdapter(server_hostname=server_hostname, max_retries=max_retries)
+    )
+    requests_session.mount("http://", adapters.HTTPAdapter(max_retries=max_retries))
```

Comment:
The `HTTPAdapter` for `https://` has a `server_hostname` and the one for `http://` doesn't. Is that a problem?

Reply:
NM. I get it now. Where did this pattern come from?

Reply:
we don't need to worry about custom TLS certs if it's not HTTPS?

Reply:
I meant the pattern for setting a hostname inside the subclassed `HTTPAdapter`. I assume it's good practice, but when I went to look for examples out there on the internet to learn more about it I wasn't very successful. Where'd the pattern come from?